include/boost/math/tools/simple_continued_fraction.hpp

```diff
@@ -29,20 +29,28 @@
 
 namespace boost::math::tools {
 
-template<typename Real, typename Z = int64_t>
+template<typename Real, typename I = int64_t>
 class simple_continued_fraction {
 public:
-    simple_continued_fraction(Real x) : x_{x} {
+    typedef I int_type;
+
+    ~simple_continued_fraction() = default;
+    simple_continued_fraction(const simple_continued_fraction&) = default;
+    simple_continued_fraction& operator =(const simple_continued_fraction&) = default;
+    simple_continued_fraction(simple_continued_fraction&&) = default;
+    simple_continued_fraction& operator =(simple_continued_fraction&&) = default;
```

Comment:
The core guidelines recommends against all of this as the compiler will generate them automatically. See: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-zero